account_bank_statement_import_online/models/online_bank_statement_provider.py

```diff
@@ -232,41 +240,47 @@ def _process_data(self, data, statement_date_since, statement_date_until):
             lines_data = []
         if not statement_values:
             statement_values = {}
-        statement = self._get_statement(statement_values, statement_date)
         self._update_statement_balance_values(
             statement_values,
             lines_data,
             statement_date_since,
             statement_date_until
         )
-        filtered_lines = self._get_filtered_lines(lines_data)
-        self._update_line_values(filtered_lines)
-        statement_values.update(
-            {
-                "line_ids": [[0, False, line] for line in filtered_lines],
-            }
+        filtered_lines = self._get_filtered_lines(
+            lines_data,
+            statement_date_since,
+            statement_date_until
         )
-        statement.write(statement_values)
+        self._update_line_values(filtered_lines)
+        self._create_statement(statement_values, statement_date, filtered_lines)
 
     @api.multi
-    def _get_statement(self, statement_values, statement_date):
+    def _create_statement(self, statement_values, statement_date, lines_data):
```

Comment:
Looking at this again - why the refactor?

Couldn we have kept `_get_statement` intact like it was, but just not call it when `lines_data` is empty, and `self.create_empty_statements` is not set?

Reply:
Well having a `_get_statement` is already part of the refactor. I Try to give each function as clear a role as possible, minimizing data passed. All according to `clean coding` principles. The main constraint is to not break inheriting modules so